docs/userguide.md

```diff
@@ -0,0 +1,151 @@
+endpoint for interacting with a list of clusters and associated metadata. If it
+helps, you can think of the cluster registry as a hosted kubeconfig file.
+However, since it's a Kubernetes API server, the cluster registry allows custom
+annotations and filtering across labels, can be used with `kubectl` and
+Kubernetes-style generated client libraries, and supports having controllers
+watch for updates.
+
+## Deploying a cluster registry
+
+The `crinit` tool allows you to deploy a cluster registry into an existing
+Kubernetes cluster. Run the following commands to download the latest nightly
+build of the tool:
+
+> Currently the tool is only released for Linux 64-bit. You'll need to built it
+> yourself if you want to use it on a different platform. See
+> [the development docs](development.md#Building-crinit).
+
+```sh
+PACKAGE=client
+LATEST=$(curl https://storage.googleapis.com/crreleases/nightly/latest)
+curl -O http://storage.googleapis.com/crreleases/nightly/$LATEST/clusterregistry-$PACKAGE.tar.gz
+tar xzf clusterregistry-$PACKAGE.tar.gz
+```
+
+### Standalone
+
+You can deploy the cluster registry as a standalone API server like so:
+
+```sh
+./crinit standalone <cluster_registry_context> --host-cluster-context=<your_cluster_context>
+```
+
+where `cluster_registry_context` is the name you want to give the new context in
+your kubecontext file that will point to your cluster registry and
+`host-cluster-context` is a context entry in your kubeconfig file. When the
+command completes, it will create an entry in your kubeconfig file for the
+cluster registry API server named with the provided `cluster_registry_context`.
+You can interact with it using kubectl:
+
+```sh
+$ kubectl get clusters --context <cluster_registry_context>
+No resources found
+$
+```
+
+### Aggregated API server
+
+> The cluster registry does not currently support delegated authn/authz, so
+> aggregation will not work smoothly: the kubectl command against the aggregator
+> will fail because the cluster registry will not be able to verify the user.
+> This is currently being addressed.
+
+You can deploy the cluster registry as an aggregated API server like so:
+
+```sh
+./crinit aggregated <cluster_registry_context> --host-cluster-context=<your_cluster_context>
+```
+
+where `host-cluster-context` is a context entry in your kubeconfig file. When
+the command completes, you will have a cluster registry running as an aggregated
+API server with the cluster identified by `host_cluster_context`. You can
+interact with it using kubectl:
+
+```sh
+$ kubectl get clusters --context <host_cluster_context>
+No resources found
+$
+```
+
+### Try it out!
+
+In these examples, `context` is either the `cluster_registry_context` if you did
+a standalone deployment, or the `host_cluster_context` if you did an aggregated
+deployment.
+
+Try creating a cluster:
+
+```sh
+kubectl apply -f - --context <context> <<EOF
+kind: Cluster
+apiVersion: clusterregistry.k8s.io/v1alpha1
```

Comment:
Are we approved for this name? AFAIU, `X.k8s.io` is only for k8s APIs, not for the k8s-style APIs. Are @kubernetes/api-approvers Ok with this?

Reply:
No, we haven't gone through that process yet. That said, I believe that the Cluster API group is also using a `k8s.io` name, as is ServiceCatalog; and this is still pre-alpha, so we can change this later without breaking any promises about support.

Filed #121 to track.

Reply:
Fair enough. Thanks for opening an issue to track this.